src/app/admin/services/controller/equipment-controller.service.ts

```diff
@@ -116,18 +117,37 @@ export class EquipmentController {
         }),
         switchMap((unavailablePeriods) => this.openBlockEquipmentModal(equipment, unavailablePeriods)),
         filter(Boolean),
-        map((period) => {
-          blockPeriod = period;
-          return this.isPeriodsIntersect(period, unavailableDates);
+        switchMap((res) => {
+          return res === 'unblock equipment'
+            ? this.unblockEquipment(equipment)
+            : this.blockEquipment(equipment, res, unavailableDates, data.action);
         }),
-        switchMap((isIntersect) => (isIntersect ? this.openOrderNotificationModal(data.action) : of(true))),
-        filter(Boolean),
-        switchMap(() => this.api.blockEquipment(equipment.id, blockPeriod)),
+        switchMap(() => this.fetchEquipments()),
         untilDestroyed(this),
       )
-      .subscribe((res) => {
-        this.notificationService.openSuccess(`${equipment.title} ${EquipmentNotification.blocked}`);
-      });
+      .subscribe();
+  }
+
+  blockEquipment(
+    equipment: Equipment,
+    period: Period,
+    unavailableDates: UnavailableDates[],
+    action: string,
+  ): Observable<unknown> {
```

Comment:
again as no rely on the emitted value and no clear return type, better to use void observable

Reply:
done